synapse/http/servlet.py

```diff
@@ -108,12 +109,11 @@ def parse_boolean_from_args(args, name, default=None, required=False):
 
 def parse_string(
     request,
-    name,
-    default=None,
-    required=False,
-    allowed_values=None,
-    param_type="string",
-    encoding="ascii",
+    name: Union[bytes, str],
+    default: Optional[str] = None,
+    required: bool = False,
+    allowed_values: Optional[Iterable[str]] = None,
+    encoding: Optional[str] = "ascii",
 ):
```

Comment:
Let's put a return type while we're at it.

```suggestion
) -> Optional[Union[bytes, str]]:
```

We usually remove the types from the docstring when adding them here.

Reply:
See continued conversation in https://github.com/matrix-org/synapse/pull/9247#discussion_r574212632